pkg/datapath/iptables/iptables.go

```diff
@@ -297,6 +297,18 @@ func (m *IptablesManager) Init() {
 			"ip6tables kernel modules could not be loaded, so IPv6 cannot be used")
 		haveIp6tables = false
 	}
+
+	ipv6Disabled, err := sysctl.Read("net.ipv6.conf.all.disable_ipv6")
```

Comment:
I'm a bit uneasy about [this part](https://github.com/torvalds/linux/blob/6e7765cb477a9753670d4351d14de93f1e9dbbd4/Documentation/networking/ip-sysctl.rst?plain=1#L1998):

> **Reading this value does not have any particular meaning**. It does not say
> whether IPv6 support is enabled or disabled. Returned value can be 1
> also in the case when some interface has ``disable_ipv6`` set to 0 and
> has configured IPv6 addresses.

If this is strictly about the `ipv6.disable=1` option, should we be looking at `/sys/module/ipv6/parameters/disable` instead?

Reply:
So in my test I set the `ipv6.disable=1` kernel parameter (which is what the bug report is about) which then results in `net.ipv6.conf.all.disable_ipv6` being set to `1`.

I'm going for this check mostly for consistency as we are already using it in another part of the codebase:

https://github.com/cilium/cilium/blob/94ea7fc93e0af5bf3ddab5b67508003ca47678cd/pkg/datapath/loader/base.go#L316

but relying on `/sys/module/ipv6/parameters/disable` would be fine as well

Reply:
> So in my test I set the `ipv6.disable=1` kernel parameter (which is what the bug report is about) which then results in `net.ipv6.conf.all.disable_ipv6` being set to `1`.
> 
> I'm going for this check mostly for consistency as we are already using it in another part of the codebase:
> 
> https://github.com/cilium/cilium/blob/94ea7fc93e0af5bf3ddab5b67508003ca47678cd/pkg/datapath/loader/base.go#L316
> 
> but relying on `/sys/module/ipv6/parameters/disable` would be fine as well


The sysctl doesn't feel like a good fit for catching `ipv6.disable=1` to me. It just controls IPv6 operations on the selected network interface(s), no? For instance, if I manually set `net.ipv6.conf.all.disable_ipv6=1`, then running the reproducer (`ip6tables -w 5 -t nat -S`) still works fine.

Reply:
Right, setting `net.ipv6.conf.all.disable_ipv6=1` will not impact the `ip6tables` command, although the inet6 addresses get removed from the interfaces:
```
vagrant@k8s1:~$ ip a | grep inet6
vagrant@k8s1:~$
```

let's switch to `/sys/module/ipv6/parameters/disable` then :+1: